File: 은빈/파이썬/[PGS]겹치는선분의길이.py
def solution(lines):
    line = set()
    
    for i in range(2):
        s1 = lines[i][0]
        e1 = lines[i][1]
        
        for j in range(i+1, 3):
            s2 = lines[j][0]
            e2 = lines[j][1]
            
            if s1 <= s2 and e1 > s2:
                if e1 < e2:
                    for dot in range(s2, e1):
                        line.add((dot, dot+1))
                        print(1, dot, dot+1)
                else:
                    for dot in range(s2, e2):
                        line.add((dot, dot+1))
                        print(2, dot, dot+1)
            elif s1 < e2 and s1 >= s2:
                if e1 < e2:
                    for dot in range(s1, e1):
                        line.add((dot, dot+1))
                        print(3, dot, dot+1)
                else:
                    for dot in range(s1, e2):
                        line.add((dot, dot+1))
                        print(4, dot, dot+1)
                        
    answer = len(line)
    
    return answer

File: 은빈/자바/[PGS]겹치는선분의길이.java
import java.util.*;

class Solution {
    public int solution(int[][] lines) {
        Map<Integer, Integer> line = new HashMap<>();

        for(int i = 0; i < 2; i++) {
            for(int j = i + 1; j < 3; j++) {
                int s = Math.max(lines[i][0], lines[j][0]);
                int e = Math.min(lines[i][1], lines[j][1]);
                
                for(int dot = s; dot < e; dot++) {
                    line.put(dot, dot + 1);
                }
            }
        }
        
        int answer = line.size();
        return answer;
    }
}